packages/filesystem/src/browser/file-dialog/file-dialog.ts

```diff
@@ -218,52 +218,22 @@ export class OpenFileDialog extends FileDialog<MaybeArray<FileStatNode>> {
         @inject(FileDialogWidget) readonly widget: FileDialogWidget
     ) {
         super(props, widget);
+        if (props.canSelectFiles !== undefined) {
+            this.widget.disableFileSelection = !props.canSelectFiles;
+        }
+        if (props.canSelectFolders !== undefined) {
+            this.widget.disableFolderSelection = !props.canSelectFolders;
```

Comment:
is there a better way to pass props from the dialog to the widget ?

Reply:
It's a good way.